File: solutions/go/space-age/2/space_age.go
package space

type Planet string

const year = 31557600

var secPerYear = map[Planet]float64{
	"Earth":   year,
	"Mercury": year * 0.2408467,
	"Venus":   year * 0.61519726,
	"Mars":    year * 1.8808158,
	"Jupiter": year * 11.862615,
	"Saturn":  year * 29.447498,
	"Uranus":  year * 84.016846,
	"Neptune": year * 164.79132,
}

func Age(seconds float64, planet Planet) float64 {
	return seconds / secPerYear[planet]
}

File: solutions/prolog/space-age/1/space_age.pl
years(AgeSec, Years):-
    Years is AgeSec / 31557600.

space_age(Planet, AgeSec, Years):-
    Planet = "Earth",
    years(AgeSec, Years).

space_age(Planet, AgeSec, Years):-
    Planet = "Mercury",
    years(AgeSec, EarthYears),
    Years is EarthYears / 0.2408467.

space_age(Planet, AgeSec, Years):-
    Planet = "Venus",
    years(AgeSec, EarthYears),
    Years is EarthYears / 0.61519726.

space_age(Planet, AgeSec, Years):-
    Planet = "Mars",
    years(AgeSec, EarthYears),
    Years is EarthYears / 1.8808158.

space_age(Planet, AgeSec, Years):-
    Planet = "Jupiter",
    years(AgeSec, EarthYears),
    Years is EarthYears / 11.862615.

space_age(Planet, AgeSec, Years):-
    Planet = "Saturn",
    years(AgeSec, EarthYears),
    Years is EarthYears / 29.447498.

space_age(Planet, AgeSec, Years):-
    Planet = "Uranus",
    years(AgeSec, EarthYears),
    Years is EarthYears / 84.016846.

space_age(Planet, AgeSec, Years):-
    Planet = "Neptune",
    years(AgeSec, EarthYears),
    Years is EarthYears / 164.79132.